Code/Source/solver/mat_models.cpp

```diff
@@ -1744,4 +1509,206 @@ void g_vol_pen(const ComMod& com_mod, const dmnType& lDmn, const double p,
   }
 }
 
+/**
+ * @brief Get the viscous PK2 stress and corresponding tangent matrix contributions for a solid
+ * with a viscous pseudo-potential model.
+ * This is defined by a viscous pseuo-potential
+ * Psi = mu/2 * tr(E_dot^2)
+ * The viscous 2nd Piola-Kirchhoff stress is given by
+ * Svis = dPsi/dE_dot 
+ *   = mu * E_dot
+ *   = mu * 1/2 * F^T * (grad(v) + grad(v)^T) * F
+ *   = mu * 1/2 * ( (F^T * Grad(v)) + (F^T * Grad(v))^T )
+ * 
+ * @tparam nsd Number of spatial dimensions
+ * @param mu Solid viscosity parameter
+ * @param eNoN Number of nodes in an element
+ * @param Nx Shape function gradient w.r.t. reference configuration coordinates (dN/dX)
+ * @param vx Velocity gradient matrix w.r.t reference configuration coordinates (dv/dX)
+ * @param F Deformation gradient matrix
+ * @param Svis Viscous 2nd Piola-Kirchhoff stress matrix
+ * @param Kvis_u Viscous tangent matrix contribution due to displacement
+ * @param Kvis_v Visous tangent matrix contribution due to velocity
+ */
+void get_visc_stress_pot(const double mu, const int eNoN, const Array<double>& Nx, const Array<double>& vx, const Array<double>& F,
+                        Array<double>& Svis, Array3<double>& Kvis_u, Array3<double>& Kvis_v) {
+
```

Comment:
I used "get" to be consistent with `get_pk2cc()`. But happy to change this (or both) to something better like "compute" or "calculate".

Reply:
@aabrown100-git It would be good to improve readability where possible, `compute` makes sense, but be careful not to change a naming convention that could cause confusion, thus change all `get_` to `compute_`.

Reply:
Got it, I changed all `get_` to `compute_` in this file. There are many other `get_`s in the repository that do a little more computing than getting, but I don't think this will cause any confusion.